internal/cluster/distribution/eks/eksadapter/node_pool_manager.go

```diff
@@ -94,3 +102,103 @@ func (n nodePoolManager) UpdateNodePool(
 func generateNodePoolStackName(clusterName string, poolName string) string {
 	return "pipeline-eks-nodepool-" + clusterName + "-" + poolName
 }
+
+// ListNodePools is for listing node pools from CloudsetFormation and NodePoolLabelSets
+func (n nodePoolManager) ListNodePools(
+	ctx context.Context,
+	c cluster.Cluster,
+	st eks.SecretStore,
+	dcf kubernetes.DynamicClientFactory,
+) ([]eks.NodePool, error) {
+	// CloudsetFormation
+	sessionFactory := workflow.NewAWSSessionFactory(st)
+	client, err := sessionFactory.New(c.OrganizationID, c.SecretID.String(), c.Location)
+	if err != nil {
+		return nil, err
+	}
+
+	cfClient := cloudformation.New(client)
+	describeStacksInput := cloudformation.DescribeStacksInput{}
+	stacksOutput, err := cfClient.DescribeStacks(&describeStacksInput)
+	if err != nil {
+		return nil, err
+	}
+
+	var relevantStacks []*cloudformation.Stack
+	for _, stack := range stacksOutput.Stacks {
+		// TODO: Later a better filtering method will be needed
+		if strings.HasPrefix(*stack.StackName, "pipeline-eks-nodepool-"+c.Name) {
+			relevantStacks = append(relevantStacks, stack)
+		}
+	}
+
+	stackMap := map[string]*cloudformation.Stack{}
+	for _, stack := range relevantStacks {
+		stackMap[*stack.StackName] = stack
+	}
+
+	// NodePoolLabelSets
+	clusterClient, err := dcf.FromSecret(ctx, c.SecretID.String())
+	if err != nil {
+		return nil, err
+	}
+
+	manager := npls.NewManager(clusterClient, global.Config.Cluster.Namespace)
+	labelSets, err := manager.GetAll()
+	if err != nil {
+		return nil, err
+	}
+
+	// Aggregate NodePoolLabelSets and CloudsetFormation Stacks
+	var nodePools []eks.NodePool
+	for labelSetName, labelSet := range labelSets {
+		stack := stackMap[labelSetName]
```

Comment:
This could be nil if the label name cannot be found in the map which would result in a panic on the next two lines. Maybe check if the value exists in the map: `stack := stackMap[labelSetName]`